inference-engine/src/mkldnn_plugin/nodes/non_max_suppression.cpp

```diff
@@ -20,82 +22,156 @@ class NonMaxSuppressionImpl: public ExtLayerBase {
+
+            const TensorDesc &scoresTensorDesc = layer->insData[NMS_SCORES].lock()->getTensorDesc();
+            if (scoresTensorDesc.getPrecision() != Precision::FP32 && scoresTensorDesc.getPrecision() != Precision::BF16)
+                THROW_IE_EXCEPTION << logPrefix << "has unsupported 'scores' input precision: " << scoresTensorDesc.getPrecision();
+
+            const SizeVector &scores_dims = scoresTensorDesc.getDims();
+            const size_t &numClasses = scores_dims[1];
+            if (scores_dims.size() != 3)
+                THROW_IE_EXCEPTION << logPrefix << "has unsupported 'scores' input rank: " << scores_dims.size();
+
+            if (numBatches != scores_dims[0])
+                THROW_IE_EXCEPTION << logPrefix << " num_batches is different in 'boxes' and 'scores' inputs";
+            if (numBoxes != scores_dims[2])
+                THROW_IE_EXCEPTION << logPrefix << " num_boxes is different in 'boxes' and 'scores' inputs";
+
+            auto dimsCheck = [&](const TensorDesc &td, const std::string inputName) {
+                const SizeVector &dims = td.getDims();
+                if (dims.size() != 0 && dims.size() != 1)
+                    THROW_IE_EXCEPTION << logPrefix << "has unsupported '" << inputName << "' input rank: " << dims.size();
+                if (dims.size() == 1)
+                    if (dims[0] != 1)
+                        THROW_IE_EXCEPTION << logPrefix << "has unsupported '" << inputName << "' input 1st dimension size: " << dims[0];
+            };
+
+            numFiltBox.resize(numBatches);
+            for (size_t i = 0; i < numFiltBox.size(); i++)
+                numFiltBox[i].resize(numClasses);
 
             if (layer->insData.size() > 2) {
-                if (layer->insData[NMS_MAXOUTPUTBOXESPERCLASS].lock()->getTensorDesc().getPrecision() != Precision::I32)
-                    THROW_IE_EXCEPTION << layer->name << " Incorrect 'max_output_boxes_per_class' input precision. Only I32 is supported!";
-                SizeVector max_output_boxes_per_class_dims = layer->insData[NMS_MAXOUTPUTBOXESPERCLASS].lock()->getTensorDesc().getDims();
-                if (max_output_boxes_per_class_dims.size() && max_output_boxes_per_class_dims[0] != 1)
-                    THROW_IE_EXCEPTION << layer->name << " 'max_output_boxes_per_class' should be scalar";
+                const TensorDesc &maxOutPerClassTensorDesc = layer->insData[NMS_MAXOUTPUTBOXESPERCLASS].lock()->getTensorDesc();
+                const Precision maxOutPerClassPrec = maxOutPerClassTensorDesc.getPrecision();
+                if (maxOutPerClassPrec != Precision::I16 && maxOutPerClassPrec != Precision::U8 && maxOutPerClassPrec != Precision::I8 &&
+                    maxOutPerClassPrec != Precision::U16 && maxOutPerClassPrec != Precision::I32 && maxOutPerClassPrec != Precision::U32 &&
+                    maxOutPerClassPrec != Precision::I64 && maxOutPerClassPrec != Precision::U64)
```

Comment:
Seems, it can be replaced on maxOutPerClassPrec.is_float()

Reply:
I believe we should compare with all supported precision because this will help avoid the situation when will be added new type not float precision that we don't validate in that ```if```